templates/app-esm/package.json

```diff
@@ -0,0 +1,3 @@
+{
+    "type": "module"
```

Comment:
All good, but the generated project prints out:

```
Suites:   ​3 passed​, ​3 of 3 completed​
Asserts:  ​​​3 passed​, ​of 3​
​Time:​   ​920.203ms​
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |       0 |        0 |       0 |       0 |                   
----------|---------|----------|---------|---------|-------------------
```

I recall that there is a configuration to add to tap to let it work

Could you check it?

Reply:
Thank you for pointing this out, I've tested the other two templates `app` and `app-ts` and they indeed print the summary properly, not sure what is the issue here, but I'll attempt to check `node-tap` docs.

Reply:
After digging through `node-tap` issues and experimenting for a while, it seems like `nyc` doesn't work well with es modules, relevant issues: https://github.com/tapjs/node-tap/issues/735 https://github.com/tapjs/node-tap/issues/813.

Anyway, there seems to be https://github.com/istanbuljs/esm-loader-hook which can be used configured to show coverage for ESM based projects. I've modified the PR accordingly to use this loader.

I'm not sure if this is the optimal solution, but it seems to get the job done. Hopefully `node-tap` will improve this scenario in an upcoming release 🤕

Reply:
Great work!

```
Suites:   3 passed, 3 of 3 completed
Asserts:  3 passed, of 3
Time:   952.563ms
---------------------|---------|----------|---------|---------|-------------------
File                 | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s
---------------------|---------|----------|---------|---------|-------------------
All files            |     100 |      100 |     100 |     100 |
 ciao                |     100 |      100 |     100 |     100 |
  app.js             |     100 |      100 |     100 |     100 |
 ciao/plugins        |     100 |      100 |     100 |     100 |
  sensible.js        |     100 |      100 |     100 |     100 |
  support.js         |     100 |      100 |     100 |     100 |
 ciao/routes         |     100 |      100 |     100 |     100 |
  root.js            |     100 |      100 |     100 |     100 |
 ciao/routes/example |     100 |      100 |     100 |     100 |
  index.js           |     100 |      100 |     100 |     100 |
---------------------|---------|----------|---------|---------|-------------------
```